new/src/hooks/useCountdown.tsx

```diff
@@ -0,0 +1,32 @@
+import { useEffect, useState } from "react";
+
+const useCountdown = (targetDate: string) => {
+  const countDownDate = new Date(targetDate).getTime();
+
+  const calculateTimeLeft = () => {
+    const now = new Date().getTime();
+    const diff = Math.max(countDownDate - now, 0);
+
+    let timeLeft = {
+      days: Math.floor(diff / (1000 * 60 * 60 * 24)),
+      hours: Math.floor((diff % (1000 * 60 * 60 * 24)) / (1000 * 60 * 60)),
+      minutes: Math.floor((diff % (1000 * 60 * 60)) / (1000 * 60)),
+    };
+
+    return timeLeft;
+  };
+
+  const [timeLeft, setTimeLeft] = useState(calculateTimeLeft);
```

Comment:
Please wrap the function declaration in a `useCallback` hook. Alternatively, you make move this function outside the `useCountdown` hook declaration.